src/server/pathway-commons/search/index.js

```diff
@@ -5,7 +5,7 @@ const getHGNCData = require('./hgnc');
 const sanitize = (s) => {
   // Escape (with '\'), to treat them literally, symbols, such as '*', ':', or space, 
   // which otherwise play special roles in a Lucene query string.
-  return s.replace(/([\!\*\+\-\&\|\(\)\[\]\{\}\^\~\?\:\/\\"\s])/g, '\\$1')
+  return s.replace(/([\!\*\+\-\&\|\(\)\[\]\{\}\^\~\?\:\/\\"\s])/g, '\\\\$1')
```

Comment:
so could you give an example of what the query will look like after this?
the goal is that backslashes should be added to the query?

Reply:
For input 'mu-type opioid' the 'queries' would be the array:

[ '(name:mu\\\\-type\\\\ opioid) OR (name:*mu\\\\-type\\\\ opioid*) OR (name:*mu\\\\-type* AND name:*opioid*)',
  '(name:*mu\\\\-type* OR name:*opioid*)',
  'mu-type opioid' ]

which, in turn, the 'pathway-common' npm package search function code interprets as:
['(name:mu\\-type\\ opioid) OR (name:*mu\\-type\\ opioid*) OR (name:*mu\\-type* AND name:*opioid*)',
...
]

Reply:
So if the pathway-commons npm package does this already, should we just not use sanitize? 

Does it work if you just make sanitize return the original string?

Reply:
I disagree with line #8 change - should stay '\\$1', which e.g., replaces:

- ' ' with '\ '
- '\' with '\\'
- ':' with '\:'
- '  ' with '\ \ '
- etc.

which was correct 
(and pathway-commons js client should not do any other tricks but URL-encoding query parameters...)

Reply:
So then it should be changed in the pathway-commons module.

Reply:
I'm suggesting that 'pathway-commons' client does tricks.

Try ['query-string' npm package](https://www.npmjs.com/package/query-string)  function 'stringify': 

```js
var stringify = require('query-string').stringify;
var s = '(name:mu\\-type\\ opioid)';
stringify({q: s}) 

// returns q=(name:mu\-type\ opioid)
```

Reply:
https://github.com/PathwayCommons/cpath2-client/blob/master/src/search.js#L102  Here is where pathway-commons module escapes lucene syntax

Reply:
maybe there is some weird behaviour where qs.stringify and the escapeLucene functions are both being run multiple times on the query...  maybe it might make sense to move away from using the pathway-commons npm package.

Reply:
Max and Igor's proposed new string:

s.replace(/([\/\\!*+-&":|()\[\]{}\^\~\?\s])/g, '\\$1')

Reply:
Manfred lib changes:
- add escape spaces for escapeLucene
- do not escapeLucene if toggle is undefined

https://github.com/PathwayCommons/cpath2-client/blob/master/src/search.js#L102

Reply:
app changes:
- needs to send toggle to the lib to determine if the toggle should be set to true or not

Reply:
We've (with @maxkfranz)  got an idea; see PathwayCommons/cpath2-client#12.